HW1/P2/P2.txt

```diff
@@ -1 +1,4 @@
 # Your discussion here
+From the graph in P2a, it is clear that the number of mandelbrot iterations varies distinctly based on its coordinates. To balance the amount of work across all the partitions, my strategy is to randomly shuffle the list of input coordinates. With such a large number of data points (4000000) randomly sorted into a small number of partitions (100), it is likely that each partition has a similar percentage of coordinates with high versus low iterations. From the graph in P2b, the work is balanced really well, with total iterations in the range of [1960000, 2100000], a spread of only 250000. In contrast, the total iterations in P2a_hist covers the a wider range of [0,9000000]
```

Comment:
I also randomly shuffled the input coordinates. One way that I had also thought about after submission was to uniformly map the high iteration points to partitions, but not sure if the preprocessing cost of that is a bottleneck